LibreNMS/Data/Graphing/GraphParameters.php

```diff
@@ -0,0 +1,265 @@
+        $this->prev_from = $this->from - $this->period;
+
+        $this->inverse = ! empty($vars['inverse']);
+        $this->in = $this->inverse ? 'out' : 'in';
+        $this->out = $this->inverse ? 'in' : 'out';
+    }
+
+    public function visible(string $element): bool
+    {
+        return $this->visibleElements[$element] ?? true;
+    }
+
+    public function all(): array
+    {
+        $variables = get_object_vars($this);
+
+        // legacy compat
+        $variables['nototal'] = ! $this->visibleElements['total'];
+        $variables['nodetails'] = ! $this->visibleElements['details'];
+        $variables['noagg'] = ! $this->visibleElements['aggregate'];
+        $variables['title'] = $this->visibleElements['title'];
+
+        return $variables;
+    }
+
+    public function toRrdOptions(): array
+    {
+        $options = ['--start', $this->from, '--end', $this->to, '--width', $this->width, '--height', $this->height];
+
+        // image must fit final dimensions
+        if ($this->full_size) {
+            $options[] = '--full-size-mode';
+        }
+
+        if ($this->imageFormat === ImageFormat::svg) {
+            $options[] = '--imgformat=SVG';
+            if ($this->width < self::MEDIUM) {
+                array_push($options, '-m', 0.75, '-R', 'light');
+            }
+        }
+
+        // set up fonts
+        array_push($options, '--font', 'LEGEND:' . $this->font_size . ':' . $this->font);
+        array_push($options, '--font', 'AXIS:' . ($this->font_size - 1) . ':' . $this->font);
+        array_push($options, '--font-render-mode', 'normal');
+
+        // set up colors
+        foreach ($this->graphColors() as $name => $color) {
+            array_push($options, '-c', $name . '#' . $color);
+        }
+
+        // set up scaling scaling
+        if (! $this->scale_min && ! $this->scale_max) {
+            $options[] = '--alt-autoscale-max';
+            if ($this->scale_rigid === null) {
+                $this->scale_rigid = true;
+            }
+        }
+        if ($this->scale_min) {
+            array_push($options, '-l', $this->scale_min);
+        }
+        if ($this->scale_max) {
+            array_push($options, '-u', $this->scale_max);
+        }
+        if ($this->scale_rigid) {
+            $options[] = '--rigid';
+        }
+
+        if ($this->sloped) {
+            $options[] = '--slope-mode';
+        }
+
+        // remove all text, height is too small
+        if ($this->height < self::TINY) {
+            $options[] = '--only-graph';
+        }
+
+        if (! $this->visible('legend')) {
+            $options[] = '-g';
+        }
+
+        if ($this->visible('title') && $this->title) {
+            $options[] = "--title='" . Clean::alphaDashSpace($this->title) . "'";
+        }
+
+        return $options;
+    }
+
+    public function __toString(): string
+    {
+        return implode(' ', $this->toRrdOptions());
+    }
+
+    public function parseAtTime(string|int $time): int
```

Comment:
This function/logic should probably live in a date/time class?

Reply:
@Jellyfrog sounds good to me